app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt

```diff
@@ -74,16 +87,26 @@ class HistoryFragment : Fragment(), BackHandler {
         setHasOptionsMenu(true)
     }
 
+    fun deleteHistoryItems(items: List<HistoryItem>) {
+        lifecycleScope.launch {
+            val storage = context?.components?.core?.historyStorage
+            for (item in items) {
+                storage?.deleteVisit(item.url, item.visitedAt)
+            }
+            reloadData()
+        }
+    }
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
         lifecycleScope.launch { reloadData() }
-    }
 
-    override fun onStart() {
-        super.onStart()
-        getAutoDisposeObservable<HistoryAction>()
-            .subscribe(this::handleNewHistoryAction)
+        historyStore.observe(view) {
+            MainScope().launch {
```

Comment:
Landed a fix here for `SearchFragment` to use lifecylcescope: https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt#L170

We need to apply this here as well for now. We're working on a solution (and better API) in A-C: https://github.com/mozilla-mobile/android-components/issues/3803